### PR TITLE
snag and roll jets

### PR DESCRIPTION
**Pull request by litlep-nibbyt**
use flog for string interpolation to print out missing jet information.

### CODE REVIEW

File: rust/sword/src/site.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+pub struct Site {
+    pub battery: Noun,    // battery
+    pub context: Noun,    // context
+    pub jet: Option<Jet>, // jet driver
+    pub path: Noun,       // label
+}
+
+impl Site {
+    /// Prepare a locally cached gate to call repeatedly.
+    pub fn new(ctx: &mut Context, core: &mut Noun) -> Site {
```
Comment (philipquirk) [inline rust/sword/src/site.rs:22]: why is core mut? is this necessary?

**Review by philipquirk — APPROVED** [review]
This is great. I have one question which is why is core mut inside a Site? I assume its necessary but I don't see why it would ever be changed.

File: rust/sword/src/site.rs (hunk 1 above)
Comment (eamsden) [reply rust/sword/src/site.rs:22]: Yes unifying equality pollutes a lot of stuff with `&mut`

File: rust/sword/src/site.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+/// Slam a cached call site.
+pub fn site_slam(ctx: &mut Context, site: &Site, sample: Noun) -> Result {
+    let subject = T(&mut ctx.stack, &[site.battery, sample, site.context]);
+    if site.jet.is_some() {
+        let jet = site.jet.unwrap();
+        jet(ctx, subject)
+    } else {
+        Ok(interpret(ctx, subject, site.battery)?)
+    }
+}
```
Comment (eamsden) [inline rust/sword/src/site.rs:73]: This would be better as an impl method on the `Site` struct. But that need not block merge for now.

**Review by eamsden — APPROVED** [review]

**Pull request merged**